src/Discovery/DiscoveryDetails/DataDownloadList/ActionButtons/ActionButtons.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react';
+import { Col, Row, Button } from 'antd';
+import HandleDownloadManifestClick from './DownloadUtils/HandleDownloadManifestClick';
+import DownloadAllModal from './DownloadAllModal/DownloadAllModal';
+import './ActionButtons.css';
+import DownloadAllFiles from './DownloadUtils/DownloadAllFiles';
+import DownloadJsonFile from './DownloadUtils/DownloadJsonFile';
+
+const ActionButtons = ({ discoveryConfig, resourceInfo }): JSX.Element => {
+  const [downloadStatus, setDownloadStatus] = useState({
+    inProgress: false,
+    message: { title: '', content: <React.Fragment />, active: false },
+  });
+  const studyIDs = [resourceInfo?.study_id];
+
+  return (
+    <div className='discovery-modal_buttons-row' data-testid='actionButtons'>
+      <DownloadAllModal
+        downloadStatus={downloadStatus}
+        setDownloadStatus={setDownloadStatus}
+      />
+      <Row className='row'>
+        {/*
+        <Col flex='1 0 auto'>
+          <Button className='discovery-action-bar-button'>
+            Download <br />
+            Variable-Level Metadata
+          </Button>
+        </Col>
+        */}
+        {discoveryConfig?.features.exportToWorkspace.studyMetadataFieldName
+          && discoveryConfig?.features.exportToWorkspace.enableDownloadStudyMetadata
+          && resourceInfo?.study_metadata && (
+          <Col flex='1 0 auto'>
+            <Button
+              className='discovery-action-bar-button'
+              onClick={() => DownloadJsonFile(
+                'study-level-metadata',
+                resourceInfo.study_metadata,
+              )}
+            >
+                Download <br />
+                Study-Level Metadata
+            </Button>
+          </Col>
+        )}
+        {discoveryConfig?.features.exportToWorkspace.enableDownloadManifest && (
+          <Col flex='1 0 auto'>
+            <Button
+              className='discovery-action-bar-button'
+              onClick={() => {
+                HandleDownloadManifestClick(
+                  discoveryConfig,
+                  [resourceInfo],
+                  false,
+                );
+              }}
+            >
+              Download File Manifest
+            </Button>
+          </Col>
+        )}
+        {discoveryConfig?.features.exportToWorkspace.enableDownloadZip && (
+          <Col flex='1 0 auto'>
+            <Button
+              className='discovery-action-bar-button'
+              onClick={() => DownloadAllFiles(studyIDs, downloadStatus, setDownloadStatus)}
+            >
+              Download All Files
```

Comment:
Same as above, but this button should also obey any additional repository specific auth restrictions for data access, see how `checkFederatedLoginStatus()` and `healRequiredIDPLogic()` is being used in https://github.com/uc-cdis/data-portal/blob/master/src/Discovery/DiscoveryActionBar.tsx

Reply:
Added logic to check user is both logged in and _healLoginNeeded_ is not true. Moved logic to _ProcessHealRequiredIdentityProviderInfo.ts_ into a separate function called from _DiscoveryDetails.tsx_, to help avoid ActionsButtons.tsx from doing processing not directly related to the action buttons.

Reply:
Note: todo Dec. 7th -> address missing checkFederatedLoginStatus() here.